src/rocksdb/src/lib.rs

```diff
@@ -787,7 +812,25 @@ fn rocksdb_core_loop<K, V, M, O, IM, F>(
                                 .bincode
                                 .serialize_into::<&mut Vec<u8>, _>(&mut encode_buf, update)
                             {
-                                Ok(()) => ret.size_written += u64::cast_from(encode_buf.len()),
+                                Ok(()) => {
+                                    ret.size_written += u64::cast_from(encode_buf.len());
+                                    // calculate the diff size if the diff multiplier is present
+                                    if let Some(diff) = diff {
+                                        let encoded_len = match i64::try_from(encode_buf.len()) {
+                                            Ok(len) => len,
+                                            Err(_) => {
+                                                let _ =
+                                                    response_sender.send(Err(Error::ValueError(
+                                                        "encoded value length too large"
+                                                            .to_string(),
+                                                    )));
+                                                return;
+                                            }
+                                        };
```

Comment:
in `upsert::types` we just `.expect()`, i'd do that here to! we are just making llvm work too hard with this!